File: Gemini 1.5 Pro/Python/2-3.py
# Get the names of the input and output files
input_filename = input("Enter the name of the input file: ")
output_filename = input("Enter the name of the output file: ")

# Open the input file for reading ("r") and the output file for writing ("w")
try:
    with open(input_filename, "r") as input_file, open(output_filename, "w") as output_file:
        # Read the content of the input file line by line
        for line in input_file:
            # Write each line to the output file
            output_file.write(line)

    print(f"Successfully copied content from '{input_filename}' to '{output_filename}'")

except FileNotFoundError:
    print(f"Error: Input file '{input_filename}' not found.")
except Exception as e:
    print(f"An error occurred: {e}")

File: Gemini 1.5 Pro/C++/2-3.c
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

int main() {
  string line;

  // Create input and output file streams
  ifstream inputFile("input.txt"); 
  ofstream outputFile("output.txt");

  // Check if input file opened successfully
  if (inputFile.is_open()) {
    // Read from input file and write to output file
    while (getline(inputFile, line)) {
      outputFile << line << endl;
    }

    // Close both files
    inputFile.close();
    outputFile.close();

    cout << "File read and write operation successful." << endl;
  } else {
    cout << "Unable to open input file." << endl;
  }

  return 0;
}

